diff --git a/src/libhelix-aac/CMakeLists.txt b/src/libhelix-aac/CMakeLists.txt
new file mode 100644
index 0000000..24ffa9d
--- /dev/null
+++ b/src/libhelix-aac/CMakeLists.txt
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.10)
+project(libhelix-aac)
+set(CMAKE_C_STANDARD 99)
+
+add_definitions(-DUSE_DEFAULT_STDLIB)
+
+file(GLOB SOURCES CONFIGURE_DEPENDS
+    "${PROJECT_SOURCE_DIR}/*.c"
+)
+file(GLOB HEADERS CONFIGURE_DEPENDS
+    "${PROJECT_SOURCE_DIR}/*.h"
+)
+
+if(BUILD_SHARED_LIBS)
+    add_library(helix-aac SHARED ${SOURCES} ${HEADERS})
+else()
+    add_library(helix-aac STATIC ${SOURCES} ${HEADERS})
+endif()
+
+install(FILES ${HEADERS}
+    DESTINATION include
+)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+install(TARGETS helix-aac
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+)
diff --git a/src/libhelix-aac/aaccommon.h b/src/libhelix-aac/aaccommon.h
index 804337c..686ad18 100644
--- a/src/libhelix-aac/aaccommon.h
+++ b/src/libhelix-aac/aaccommon.h
@@ -46,8 +46,14 @@
 #ifndef _AACCOMMON_H
 #define _AACCOMMON_H
 
-#include <Arduino.h>
-#include <pgmspace.h>
+#ifdef ARDUINO
+  #include <Arduino.h>
+  #include <pgmspace.h>
+#else
+    #define PROGMEM
+    #define pgm_read_byte(addr) (*(const unsigned char *)(addr))
+    #define pgm_read_word(addr) (*(const unsigned short *)(addr))
+#endif
 
 // Can't fit in ESP8266 RAM
 #ifndef ESP8266
diff --git a/src/libhelix-aac/statname.h b/src/libhelix-aac/statname.h
index a27f04d..286029d 100644
--- a/src/libhelix-aac/statname.h
+++ b/src/libhelix-aac/statname.h
@@ -50,16 +50,8 @@
  * all the C functions and global variables will be mangled by the preprocessor
  *   e.g. void DCT4(...) becomes void raac_DCT4(...)
  */
-#define STAT_PREFIX		raac
 
-#define STATCC1(x,y,z)	STATCC2(x,y,z)
-#define STATCC2(x,y,z)	x##y##z  
-
-#ifdef STAT_PREFIX
-#define STATNAME(func)	STATCC1(STAT_PREFIX, _, func)
-#else
-#define STATNAME(func)	func
-#endif
+#define STATNAME(func) func
 
 /* these symbols are common to all implementations */
 #define AllocateBuffers			STATNAME(AllocateBuffers)
